lib/services/auth/modelFilters/statement.js

```diff
@@ -28,9 +29,19 @@ const hasReadMineScope = (clientScopes) => {
   );
 };
 
+const hasWriteScope = (clientScopes) => {
+  if (!clientScopes) {
+    return false;
+  }
+  return includes(clientScopes, scopes.XAPI_ALL, scopes.XAPI_STATEMENTS_WRITE);
+};
+
 export default async ({ actionName, authInfo }) => {
   const tokenType = getTokenTypeFromAuthInfo(authInfo);
-  if (actionName !== 'view') throw new NoAccessError();
+
+  if (actionName !== 'view' && !hasWriteScope(get(authInfo, 'client.scopes'))) {
```

Comment:
Will this inadvertently enable adds, edits and deletions on xAPI statements via the Restify endpoints?